### DISCUSSION TITLE
Where can i ask for help when coding in tukaan?

### DISCUSSION THREAD

**Discussion by lmjaedentai** [tukaan/tukaan#91] [Question] [answered]
Just a minor suggestion, a discord server for us to ask for help when we met error / facing problem, like what we did in stackoverflow.

For example, I want to ask:
> why this error occur although i fully copy the example from docs?
```py
import tukaan
from tukaan import Label

app = tukaan.App(title="My window")
app.width = 1000
app.height= 1000

Label(app, text = "Hello, World!").layout.grid()
app.run()
```
error:
```py
Traceback (most recent call last):
  File "d:\Desktop\coding\sandbox\main.py", line 7, in <module>
    Label(app, text = "Hello, World!").layout.grid()
AttributeError: 'Label' object has no attribute 'layout'
```
The example given in docs also cannot be used sometimes, for example `from tukaan import Label` arent available in docs, causing user have to add it by themself to avoid NameError.

**Comment by Moosems** [comment, marked as answer]
@lmjaedentai Hello! To ask for help simply open an issue or ask a question in discussions. Since it is easier to just use these than to force people to get additional software, we plan on sticking with that. As for the tukaan docs we have recently changed the way that tukaan works and does layouting meaning that we must rewrite the docs. A working sample is:

```python
import tukaan
from tukaan import Label

app = tukaan.App(title="My window")
app.width = 1000
app.height= 1000

Label(app, text = "Hello, World!").grid()
app.run()
```

Hope these answer your questions!

**Reply by rdbende** [reply]
Actually, I think it'd be better to have a separate place for asking questions, and giving help. GitHub issues is good for bug reports and proposals, but not for asking help. I often find it difficult to find relevant entries, and if they get cluttered with other questions, it doesn't help the overall transparency of the project.

**Reply by Moosems** [reply]
So where should we do it?

**Reply by rdbende** [reply]
@Moosems Thinking on it. TBH I've never asked questions on SO, Reddit, or on any other platform in my life, so I don't have much knowledge of how that works. Maybe if the documentation is detailed and easy to understand, there's less need for people to ask questions (Tkinter sucks in that, it has no official docs).

@lmjaedentai Anyways, I'll update the documentation, to reflect the changes.

**Reply by Moosems** [reply]
@rdbende What if we learned some JavaScript and added a chat to the website?

**Reply by lmjaedentai** [reply]
@rdbende @Moosems thanks for your help!

**Reply by Moosems** [reply]
We could use IRC now that I think about it

**Reply by lg-nineoneninethreenine** [reply]
You can use the Matrix protocol too. It works with different clients, now that I think of it.
